docs/CodeTransformation.md

```diff
@@ -0,0 +1,94 @@
+---
+id: code-transformation
+title: Code Transformation
+---
+
+Jest runs the code in your project as JavaScript, but if you use some syntax not supported by Node.js out of the box (such as JSX, types from TypeScript, Vue templates etc.) then you'll need to transform that code into plain JavaScript, similar to what you would do when building for browsers.
+
+Jest supports this via the [`transform` configuration option](Configuration.md#transform-objectstring-pathtotransformer--pathtotransformer-object).
+
+A transformer is a module that provides a synchronous function for transforming source files. For example, if you wanted to be able to use a new language feature in your modules or tests that aren't yet supported by Node, you might plug in one of many compilers that compile a future version of JavaScript to a current one.
+
+Jest will cache the result of a transformation and attempt to invalidate that result based on a number of factors, such as the source of the file being transformed and changing configuration.
+
+## Defaults
+
+Jest ships with one transformer out of the box - `babel-jest`. It will automatically load your project's Babel configuration and transform any file matching the following RegEx: `/\.[jt]sx?$/` meaning any `.js`, `.jsx`, `.ts` and `.tsx` file. In addition, `babel-jest` will inject the Babel plugin necessary for mock hoisting talked about in [ES Module mocking](ManualMocks.md#using-with-es-module-imports).
+
+If you override the `transform` configuration option `babel-jest` will no longer be active, and you'll need to add it manually if you wish to use Babel.
+
+## Writing custom transformers
+
+You can write you own transformer. The API of a transformer is as follows:
+
+```ts
+interface Transformer {
+  canInstrument?: boolean;
```

Comment:
maybe worth to add a comment it's about source map coverage? I've seen people being confused about what "instrument" is

Reply:
[`e2f0881` (#10834)](https://github.com/facebook/jest/pull/10834/commits/e2f08810c88ee32ed78fc126cd2d999ecc9b8467)

Reply:
this prop is an optional prop. I think the default value should be mentioned here. Correct me if I'm wrong that if custom transformer doesn't define value for this prop, Jest will instrument code automatically using Babel ?